aws_lambda_powertools/logging/formatter.py

```diff
@@ -152,7 +152,11 @@ def format(self, record: logging.LogRecord) -> str:  # noqa: A003
         """Format logging record as structured JSON str"""
         formatted_log = self._extract_log_keys(log_record=record)
         formatted_log["message"] = self._extract_log_message(log_record=record)
-        formatted_log["exception"], formatted_log["exception_name"] = self._extract_log_exception(log_record=record)
+        # exception and exception_name fields can be added as extra key
+        # in any log level, we try to extract and use them first
+        extracted_exception, extracted_exception_name = self._extract_log_exception(log_record=record)
+        formatted_log["exception"] = formatted_log.get("exception", extracted_exception)
+        formatted_log["exception_name"] = formatted_log.get("exception_name", extracted_exception_name)
```

Comment:
I liked this solution. We could use an `if condition` to test if the keys already exist, and if not, try to extract from the exception, but that works fine too. Just out of curiosity, I ran this solution and the solution using `if condition` and we didn't get any time penalty running this. The Logger utility is widely used by our customers, so every millisecond counts in its execution.

Reply:
cheers, adding an `if condition` was my first idea, but I discard it because I had to make the decision if I should check just one or both keys and what to do if only one of the keys was provided. 

Also this approach is not affecting existing performance since `_extract_log_exception` was always called before, but please feel free to improve it further. Regards